src/dbnode/storage/index.go

```diff
@@ -748,6 +748,7 @@ func (i *nsIndex) Flush(
 
 	builderOpts := i.opts.IndexOptions().SegmentBuilderOptions()
 	builder, err := builder.NewBuilderFromDocuments(builderOpts)
+	defer builder.Close()
```

Comment:
Need to move this after the `err != nil { return err }` since if `err != nil` then `builder` will be nil and calling `builder.Close()` will cause a nil ptr panic.